src/main/java/uk/gov/justice/hmpps/nomis/datacompliance/repository/jpa/repository/DuplicateOffenderRepository.java

```diff
@@ -69,4 +69,150 @@
             ")",
             nativeQuery = true)
     List<DuplicateOffender> getOffendersWithMatchingLidsNumbers(String offenderNo);
+
+    /**
+     * The following query identifies potential duplicate offender records
+     * based on matching personal details. Key components of this query are
+     * as follows:
+     *
+     * - The Jaro Winkler algorithm is used to fuzzy match details to avoid
+     *   misspellings or typos affecting the result
+     *
+     * - The match is points based, with certain fields having a higher
+     *   weighting
+     *
+     * - All offender aliases are taken into account and for each field the
+     *   values with maximum similarity are used towards the points calculation
+     *
+     * - The weightings have been chosen based on empirical evidence
+     */
+    @Query(value =
+            "WITH OFFENDER_MATCH_DETAILS AS (" +
+            "    SELECT *" +
+            "    FROM MV_OFFENDER_MATCH_DETAILS" +
+            "    WHERE OFFENDER_ID_DISPLAY = :offenderNo" +
+            "), " +
+            "SIMILARITY AS (" +
+            "    SELECT" +
+            "        O.OFFENDER_ID_DISPLAY AS OFFENDER_ID_DISPLAY," +
+            "        (CASE WHEN OMD.AVG_HEIGHT = 0 THEN 0 ELSE ((OMD.AVG_HEIGHT - ABS(O.AVG_HEIGHT - OMD.AVG_HEIGHT)) * 100 / OMD.AVG_HEIGHT) END) AS HEIGHT_SIMILARITY," +
+            "        UTL_MATCH.JARO_WINKLER_SIMILARITY (O.LAST_NAME_KEY, OMD.LAST_NAME_KEY) AS LAST_NAME_SIMILARITY," +
+            "        UTL_MATCH.JARO_WINKLER_SIMILARITY (O.FIRST_NAME_KEY, OMD.FIRST_NAME_KEY) AS FIRST_NAME_SIMILARITY," +
+            "        (CASE WHEN ABS(MONTHS_BETWEEN(O.BIRTH_DATE, OMD.BIRTH_DATE)) >= 120 THEN 0" +
+            "         ELSE UTL_MATCH.JARO_WINKLER_SIMILARITY (O.BIRTH_DATE, OMD.BIRTH_DATE) END" +
+            "        ) AS BIRTH_DATE_SIMILARITY," +
+            "        UTL_MATCH.JARO_WINKLER_SIMILARITY (O.BIRTH_PLACE, OMD.BIRTH_PLACE) AS BIRTH_PLACE_SIMILARITY," +
+            "        UTL_MATCH.JARO_WINKLER_SIMILARITY (O.STREET, OMD.STREET) AS STREET_SIMILARITY," +
```

Comment:
The ones below feel less useful given people move - interestingly to see how effective they are

Reply:
That is fair, but they did seem to be particularly helpful in the subset of offenders we looked at to form the weightings.  When we are able to run all the different variants of the duplicate detection alongside each other (image, id, analytical platform linking and this), we'll start to spot where the gaps are in each method.